File: 프로그래머스/unrated/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerine):
    answer = 0
    d = dict()
    
    for i in list(set(tangerine)):
        d[i]=0
    for t in tangerine:
        d[t]+=1
    
    new = sorted(d.items(),key=lambda x:x[1], reverse=True)
    for n in new:
        if k>0:
            k-=n[1]
            answer+=1
        else:
            break
        
        
    return answer

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.js
function solution(k, tangerine) {
    var answer = 0;
    
    let obj = {}
    
    // object에 크기별 개수 저장
    tangerine.forEach((item)=>{
        obj[item] = ++obj[item] || 1        
    })
    
    // object의 개수 내림차순 
    const sortedObj = Object.values(obj).sort((x,y)=>y-x)
    
    // 가지 수 저장
    let cnt = 0
    
    for(let num of sortedObj){
        ++answer
        cnt += num
        
        // k개 보다 많으면 멈춤
        if(k<=cnt){
            break
        }
    }

    return answer;
}